docs/USAGE-GUIDE.md

```diff
@@ -1626,6 +1626,42 @@ failure for the same **op**.
 be called for each of the **op** received in **s2n_async_pkey_fn** to
 avoid any memory leaks.
 
+### Offloading asynchronous private key operations
```

Comment:
Once you add the API for getting sig alg, you might want to mention that here since anyone implementing this will probably be thinking "how do I know what sig alg".

Reply:
Yes I agree, I will do that in said PR, I haven't yet committed to what it looks like.